### PR TITLE
Make Profiler robust to errors

### PR DESCRIPTION
**Pull request by mrocklin**
In particular we remove incomplete results from the log

### CODE REVIEW

**Comment by mrocklin** [comment]
When you have a moment @jcrist this could use your review.  In particular I'm unsure about the profile names commit.

File: dask/diagnostics/profile_visualize.py [hunk 1]
```diff
@@ -13,7 +13,7 @@
 from ..core import istask
 
 
-def pprint_task(task, keys):
+def pprint_task(task, keys, budget=60):
```
Comment (jcrist) [inline dask/diagnostics/profile_visualize.py]: This kwarg should be added to the docstring. It also might read easier as `max_task_length`, or `label_size` or something a bit more descriptive. Not a major gripe though.

File: dask/diagnostics/profile_visualize.py [hunk 2]
```diff
@@ -80,7 +92,7 @@ def get_colors(palette, funcs):
 
 
 def visualize(results, dsk, palette='GnBu', file_path=None,
-              show=True, **kwargs):
+              show=True, save=True, **kwargs):
```
Comment (jcrist) [inline dask/diagnostics/profile_visualize.py]: Would be nice if `budget` was checked for in `kwargs`. Could be unspecified, and just checked in a `kwargs.get('budget', 60)` if the signature feels a little long. Would need to be set in the calls to pprint_task as well.

File: dask/diagnostics/profile_visualize.py (hunk 1 above)
Comment (mrocklin) [inline dask/diagnostics/profile_visualize.py]: Changed to `label_size`

Comment (mrocklin) [inline dask/diagnostics/profile_visualize.py]: and added to docstring

File: dask/diagnostics/profile_visualize.py (hunk 2 above)
Comment (mrocklin) [inline dask/diagnostics/profile_visualize.py]: added `label_size` to kwargs

**Comment by jcrist** [comment]
LGTM

**Pull request merged**